presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestApplyTableScanRedirect.java

```diff
@@ -0,0 +1,193 @@
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.filter;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.tableScan;
+import static io.prestosql.sql.planner.iterative.rule.test.RuleTester.defaultRuleTester;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+
+public class TestApplyTableScanRedirect
+{
+    private static final String MOCK_CATALOG = "mock_catalog";
+    private static final String TEST_SCHEMA = "test_schema";
+    private static final String TEST_TABLE = "test_table";
+    private static final SchemaTableName sourceTable = new SchemaTableName(TEST_SCHEMA, TEST_TABLE);
+    private static final TableHandle TEST_TABLE_HANDLE = createTableHandle(new MockConnectorTableHandle(sourceTable));
+
+    private static final Session MOCK_SESSION = testSessionBuilder().setCatalog(MOCK_CATALOG).setSchema(TEST_SCHEMA).build();
+
+    private static final String sourceColumnName = "source_col";
+    private static final ColumnHandle sourceColumnHandle = new MockConnectorColumnHandle(sourceColumnName, VARCHAR);
+
+    private static final SchemaTableName destinationTable = new SchemaTableName("target_schema", "target_table");
+    private static final String destinationColumnName = "destination_col";
+    private static final ColumnHandle destinationColumnHandle = new MockConnectorColumnHandle(destinationColumnName, VARCHAR);
+
+    private static TableHandle createTableHandle(ConnectorTableHandle tableHandle)
+    {
+        return new TableHandle(
+                new CatalogName(MOCK_CATALOG),
+                tableHandle,
+                TestingTransactionHandle.create(),
+                Optional.empty());
+    }
+
+    @Test
+    public void testDoesNotFire()
+    {
+        try (RuleTester ruleTester = defaultRuleTester()) {
+            MockConnectorFactory mockFactory = createMockFactory(Optional.empty());
+            ruleTester.getQueryRunner().createCatalog(MOCK_CATALOG, mockFactory, ImmutableMap.of());
+
+            ruleTester.assertThat(new ApplyTableScanRedirect(ruleTester.getMetadata()))
+                    .on(p -> {
+                        Symbol column = p.symbol(sourceColumnName, VARCHAR);
+                        return p.tableScan(TEST_TABLE_HANDLE,
+                                        ImmutableList.of(column),
+                                        ImmutableMap.of(column, sourceColumnHandle));
+                    })
+                    .withSession(MOCK_SESSION)
+                    .doesNotFire();
+        }
+    }
+
+    @Test
+    public void doesNotFireIfNoTableScan()
+    {
+        try (RuleTester ruleTester = defaultRuleTester()) {
+            MockConnectorFactory mockFactory = createMockFactory(Optional.empty());
```

Comment:
no redirection here, so it wouldn't fire anyway (tested by previous case). We can keep this test though